staging/src/k8s.io/apimachinery/pkg/util/yaml/decoder_test.go

```diff
@@ -22,12 +22,26 @@ import (
 	"encoding/json"
 	"fmt"
 	"io"
+	"io/ioutil"
 	"math/rand"
 	"reflect"
 	"strings"
 	"testing"
 )
 
+func TestYAMLDecoder(t *testing.T) {
+	d := `---
+stuff: 1
+	test-foo: 1
+`
+	s := NewDocumentDecoder(ioutil.NopCloser(bytes.NewReader([]byte(d))))
+	b := make([]byte, len(d))
```

Comment:
Add a case where the length of the byte buffer you are reading into is larger than len(d)